### PR TITLE
Csi PX support

### PR DESCRIPTION
**Pull request by lpabon**
Support for Stork to work with PVs which where created by Portworx CSI Provisioner

### CODE REVIEW

**Comment by disrani-px** [comment]
@lpabon What is the status for this? Is this still on track for 2.3.0?

File: pkg/k8sutils/k8sutils.go [hunk 1]
```diff
@@ -49,3 +50,26 @@ func GetVolumeNamesFromLabelSelector(namespace string, labels map[string]string)
 
 	return volNames, nil
 }
+
+// GetDriverTypeFromPV returns the name of the provisioner driver managing
+// the persistent volume. Supports in-tree and CSI PVs
+func GetDriverTypeFromPV(pv *v1.PersistentVolume) (string, error) {
```
Comment (disrani-px) [inline pkg/k8sutils/k8sutils.go]: Why put this here? Doesn't look like anything but the portworx driver will require this

Comment (disrani-px) [reply pkg/k8sutils/k8sutils.go]: Also, this only really returns driver types that support snapshots (if not CSI), so the name isn't really accurate.

Comment (lpabon) [reply pkg/k8sutils/k8sutils.go]: Gotcha, good point.

**Comment by lpabon** [comment]
Tested migration from Kubernets 1.15/Porx2.1 -> Kubernetes 1.15/Porx2.1 using CSI on both systems:

```

**Comment by lpabon** [comment]
Volume snapshot also worked

Snapshot info:

```
$ kk get volumesnapshots -o yaml
apiVersion: v1
items:
- apiVersion: volumesnapshot.external-storage.k8s.io/v1
  kind: VolumeSnapshot
  metadata:
    creationTimestamp: "2019-07-17T22:27:00Z"
    generation: 2
    labels:
      SnapshotMetadata-PVName: pvc-d2a47415-1aef-428c-b998-5aee138d93a9
      SnapshotMetadata-Timestamp: "1563402420997604515"
    name: newsnap
    namespace: default
    resourceVersion: "243150"
    selfLink: /apis/volumesnapshot.external-storage.k8s.io/v1/namespaces/default/volumesnapshots/newsnap
    uid: 14c6919c-00cb-463b-8667-a3aa39eaeca7
  spec:
    persistentVolumeClaimName: mysql-data
    snapshotDataName: k8s-volume-snapshot-ff961475-a8e1-11e9-92d7-b20d36e0f6b0
  status:
    conditions:
    - lastTransitionTime: "2019-07-17T22:27:02Z"
      message: Snapshot created successfully and it is ready
      reason: ""
      status: "True"
      type: Ready
    creationTimestamp: null
kind: List
metadata:
  resourceVersion: ""
  selfLink: ""
```


PV created from snapshot:

```
apiVersion: v1
kind: PersistentVolume
metadata:
  annotations:
    pv.kubernetes.io/provisioned-by: stork-snapshot
    snapshotProvisionerIdentity: stork
  creationTimestamp: "2019-07-17T22:32:58Z"
  finalizers:
  - kubernetes.io/pv-protection
  name: pvc-14f1a812-a261-48ab-a143-f62d24de3d2d
  resourceVersion: "244068"
  selfLink: /api/v1/persistentvolumes/pvc-14f1a812-a261-48ab-a143-f62d24de3d2d
  uid: 7ed2cbe6-2a06-4b5a-b706-9d7ec7dad6cc
spec:
  accessModes:
  - ReadWriteOnce
  capacity:
    storage: "2"
  claimRef:
    apiVersion: v1
    kind: PersistentVolumeClaim
    name: pvcfromsnap
    namespace: default
    resourceVersion: "244059"
    uid: 14f1a812-a261-48ab-a143-f62d24de3d2d
  csi:
    driver: com.openstorage.pxd
    fsType: FS_TYPE_EXT4
    volumeHandle: "978231959928143642"
  persistentVolumeReclaimPolicy: Delete
  storageClassName: stork-snapshot-sc
  volumeMode: Filesystem
status:
  phase: Bound
```

**Comment by lpabon** [comment]
all tests have passed

**Pull request merged**